commons/src/main/java/com/github/globocom/viewport/commons/ViewPortPartialHelper.kt

```diff
@@ -0,0 +1,118 @@
+package com.github.globocom.viewport.commons
+
+import android.view.View
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.OrientationHelper
+import androidx.recyclerview.widget.RecyclerView
+import kotlin.math.abs
+
+object ViewPortPartialHelper {
+
+    fun findFirstPartiallyVisibleItemPosition(
+        recyclerView: RecyclerView,
+        linearLayoutManager: LinearLayoutManager,
+        threshold: Float = 1f
+    ): Int {
+        findOneVisibleChild(
+            linearLayoutManager,
+            0,
+            linearLayoutManager.childCount,
+            threshold
+        )?.let { childView ->
+            return recyclerView.getChildAdapterPosition(childView)
+        }
+        return RecyclerView.NO_POSITION
+    }
+
+    fun findLastPartiallyVisibleItemPosition(
+        recyclerView: RecyclerView,
+        linearLayoutManager: LinearLayoutManager,
+        threshold: Float = 1f
+    ): Int {
+        findOneVisibleChild(
+            linearLayoutManager,
+            linearLayoutManager.childCount - 1,
+            -1,
+            threshold
+        )?.let { childView ->
+            return recyclerView.getChildAdapterPosition(childView)
+        }
+        return RecyclerView.NO_POSITION
+    }
+
+    private fun findOneVisibleChild(
+        layoutManager: RecyclerView.LayoutManager,
+        fromIndex: Int,
+        toIndex: Int,
+        threshold: Float
+    ): View? {
+        val helper: OrientationHelper = if (layoutManager.canScrollVertically()) {
+            OrientationHelper.createVerticalHelper(layoutManager)
+        } else {
+            OrientationHelper.createHorizontalHelper(layoutManager)
+        }
+
+        val start: Int = helper.startAfterPadding
+        val end: Int = helper.endAfterPadding
+        val next = if (toIndex > fromIndex) 1 else -1
+        var index = fromIndex
+
+        while (index != toIndex) {
+            val child: View? = layoutManager.getChildAt(index)
+            val childStart: Int = helper.getDecoratedStart(child)
+            val childEnd: Int = helper.getDecoratedEnd(child)
+            if (childStart < end && childEnd > start) {
+                if (childStart >= start && childEnd <= end) {
+                    return child
+                } else {
+                    if (checkThreshold(threshold, childStart, childEnd, end, next == 1)) {
+                        return child
+                    }
+                }
+            }
+            index += next
+        }
+        return null
+    }
```

Comment:
E nessa parte ficaria meio que assim... (incluindo outras alterações que também sugiro, em relação ao `while` e aos parâmetros)
```suggestion
private fun findOneVisibleChild(
        layoutManager: RecyclerView.LayoutManager,
        threshold: Float,
        searchFromEndToStart: Boolean = false
    ): View? {
        val helper: OrientationHelper = if (layoutManager.canScrollVertically()) {
            OrientationHelper.createVerticalHelper(layoutManager)
        } else {
            OrientationHelper.createHorizontalHelper(layoutManager)
        }

        val orderedChildSearch = 0.until(layoutManager.childCount).let {
            if (searchFromEndToStart) it.reversed() else it
        }

        for (index in orderedChildSearch) {
            layoutManager.getChildAt(index)?.let {
                if (isVisiblePortionInsideThreshold(it, helper, threshold))
                    return it
            }
        }
        return null
    }
```